### ISSUE TITLE
fork project does not url encode the project name

### ISSUE THREAD

**Issue by logicminds** [NARKOZ/gitlab#289]
We need to url encode this project name if not already done so.
https://github.com/NARKOZ/gitlab/blob/master/lib/gitlab/client/projects.rb#L414

**Comment by logicminds** [comment]
Looks like there are many methods that do not encode the name.  We should just do this in the request object automatically.  Thoughts?

**Comment by asedge** [comment]
Ideally, yes we should.  I think it might end up being a rather large change to make it happen and I wasn't willing to invest the time.

**Comment by logicminds** [comment]
I think we can escape much of this work if we test to see if the string is already encoded.

```ruby

def post(path, options={})
      set_httparty_config(options)
      set_authorization_header(options)
      path = url_encode(path) unless already_encoded?(path)
      validate self.class.post(@endpoint + path, options)
end

def already_encoded?(path)
  # this probably will not work but something along this lines 
  path.contains?('%2F') unless path.contains?(/\d/)
end
```

**Comment by asedge** [comment]
You cannot URL encode the entire path.  You could probably `split(/\//)`, `map` each element with `url_encode`, then `join('/')`.  That's a crappy way to handle it but it might actually work.

**Comment by NARKOZ** [comment]
Fixed in #325
